src/app/models/Project.ts

```diff
@@ -15,68 +15,56 @@ import { AbstractModel } from "./AbstractModel";
  * A project model.
  */
 export interface ProjectInterface {
-  kind?: "Project";
-  id: Id;
-  name: Param;
+  id?: Id;
+  name?: Param;
   imageUrl?: string;
-  creatorId: Id;
+  creatorId?: Id;
   createdAt?: DateTimeTimezone | string;
   updaterId?: Id;
   updatedAt?: DateTimeTimezone | string;
   ownerId?: Id;
   description?: Description;
-  siteIds: Ids;
+  siteIds?: Ids;
 }
 
 /**
  * A project model.
  */
 export class Project extends AbstractModel implements ProjectInterface {
   public readonly kind: "Project" = "Project";
-  public readonly id: Id;
-  public readonly name: Param;
-  public readonly imageUrl: string;
-  public readonly siteIds: Ids;
-  public readonly creatorId: Id;
+  public readonly id?: Id;
+  public readonly name?: Param;
+  public readonly imageUrl?: string;
+  public readonly siteIds?: Ids;
+  public readonly creatorId?: Id;
   public readonly createdAt?: DateTimeTimezone;
   public readonly updaterId?: Id;
   public readonly updatedAt?: DateTimeTimezone;
   public readonly ownerId?: Id;
   public readonly description?: Description;
 
   constructor(project: ProjectInterface) {
-
     super(project);
 
     // TODO: most of these are redundant - reimplement with properties?
-    this.id = project.id;
-    this.name = project.name;
     this.imageUrl =
       project.imageUrl || "/assets/images/project/project_span4.png";
-    this.creatorId = project.creatorId;
     this.createdAt = project.createdAt
       ? DateTime.fromISO(project.createdAt as string, {
           setZone: true
         })
       : defaultDateTimeTimezone;
```

Comment:
What purpose does `setZone` have?

Reply:
setZone makes the luxon date object accept accept timezone data. By default, luxon and javascript dates ignore timezones

Reply:
Ah goodo